pandas/core/groupby.py

```diff
@@ -3507,6 +3507,16 @@ def transform(self, func, *args, **kwargs):
         Each subframe is endowed the attribute 'name' in case you need to know
         which group you are working on.
 
+        The current implementation imposes three requirements on f:
+
+        * f must return a value that either has the same shape as the input
+          subframe or can be broadcast to the shape of the input subframe.
```

Comment:
can you give an example here (and in the docs), for each of these? and example function that is